wger/utils/permissions.py

```diff
@@ -36,7 +36,7 @@ def has_permission(self, request, view):
         view. If it is not present, treat is as a public endpoint
         '''
         if hasattr(view, 'is_private') and view.is_private:
-            return request.user and request.user.is_authenticated()
```

Comment:
this function is being used in more places than just with your new feature, removing this affects all the other functions that use it..